src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1987,15 +2046,17 @@ UCS_PROFILE_FUNC_ALWAYS(ucs_status_t, uct_ib_mlx5_devx_reg_exported_key,
         goto out_umem_mr;
     }
 
+    ucs_assert(!(memh->super.flags & UCT_IB_MEM_MULTITHREADED));
     status = uct_ib_mlx5_devx_reg_ksm_data_contig(md,
                                                   &memh->mrs[UCT_IB_MR_DEFAULT],
-                                                  memh->address,
-                                                  (uint64_t)memh->address, 0, 0,
+                                                  memh->address, 0, 0, 0,
                                                   "exported key", &cross_mr,
                                                   &exported_lkey);
     if (status != UCS_OK) {
         return status;
     }
+    ucs_assert(exported_lkey != 0);
+    ucs_assert(cross_mr != NULL);
```

Comment:
why needed? Does it fix some valgrind error?

Reply:
Before my patch, both `exported_lkey` and `cross_mr` were only defined in the beginning of this function. After adding the logging func in the end of `uct_ib_mlx5_devx_reg_ksm_data_contig`, compilation on Ubuntu 18.04 and SLES 15 started to fail (all other platforms passed) with message that these variables can be uninitialized.

So I decided to initialize them and check that we change them by the right way by these asserts.

Reply:
Here is the log of failure:

```
/__w/1/s/contrib/../src/uct/ib/mlx5/dv/ib_mlx5dv_md.c: In function 'uct_ib_mlx5_devx_reg_exported_key':
/__w/1/s/contrib/../src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:487:17: error: 'exported_lkey' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     return mkey >> 8;
            ~~~~~^~~~
/__w/1/s/contrib/../src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:2071:14: note: 'exported_lkey' was declared here
     uint32_t exported_lkey;
              ^~~~~~~~~~~~~
/__w/1/s/contrib/../src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:2109:25: error: 'cross_mr' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     memh->cross_mr      = cross_mr;
     ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/__w/1/s/contrib/../src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:2070:29: note: 'cross_mr' was declared here
     struct mlx5dv_devx_obj *cross_mr;
                             ^~~~~~~~
```

It appeared after adding `uct_ib_mlx5_devx_ksm_log` function to `uct_ib_mlx5_devx_reg_ksm_data_contig` which isn't touch any uninitialized memory. Removing log function solves the issue.

Initialization of these variables seems the best way to solve this issue.

Reply:
I found that the issue can be solved by removing information about `mkey` from `uct_ib_mlx5_devx_ksm_log ` function. That information is already printed inside `uct_ib_mlx5_devx_reg_ksm` in case of successfull registration, so, I think it can be removed here.